AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt

```diff
@@ -0,0 +1,702 @@
+                // Low stiffness spring for the left edge so it moves slower than the right edge.
+                spring(stiffness = Spring.StiffnessVeryLow)
+            } else {
+                // Indicator moves to the left.
+                // Medium stiffness spring for the left edge so it moves faster than the right edge.
+                spring(stiffness = Spring.StiffnessMedium)
+            }
+        }
+    ) { index ->
+        tabPositions[index].left
+    }
+    val indicatorRight by transition.animateDp(
+        transitionSpec = {
+            if (initialState < targetState) {
+                // Indicator moves to the right
+                // Medium stiffness spring for the right edge so it moves faster than the left edge.
+                spring(stiffness = Spring.StiffnessMedium)
+            } else {
+                // Indicator moves to the left.
+                // Low stiffness spring for the right edge so it moves slower than the left edge.
+                spring(stiffness = Spring.StiffnessVeryLow)
+            }
+        }
+    ) { index ->
+        tabPositions[index].right
+    }
+    val color by transition.animateColor { index ->
+        if (index == 0) purple700 else green800
+    }
+    Box(
+        Modifier
+            .fillMaxSize()
+            .wrapContentSize(align = Alignment.BottomStart)
+            .offset(x = indicatorLeft)
+            .width(indicatorRight - indicatorLeft)
+            .padding(4.dp)
+            .fillMaxSize()
+            .border(
+                BorderStroke(2.dp, color),
+                RoundedCornerShape(4.dp)
+            )
+    )
+}
+
+/**
+ * Shows a tab.
+ *
+ * @param icon The icon to be shown on this tab.
+ * @param title The title to be shown on this tab.
+ * @param onClick Called when this tab is clicked.
+ * @param modifier The [Modifier].
+ */
+@Composable
+private fun HomeTab(
+    icon: ImageVector,
+    contentDescription: String?,
+    title: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        modifier = modifier
+            .clickable(onClick = onClick)
+            .padding(16.dp),
+        horizontalArrangement = Arrangement.Center,
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Icon(
+            imageVector = icon,
+            contentDescription = contentDescription
+        )
+        Spacer(modifier = Modifier.width(16.dp))
+        Text(text = title)
+    }
+}
+
+/**
+ * Shows the weather.
+ *
+ * @param onRefresh Called when the refresh icon button is clicked.
+ */
+@Composable
+private fun WeatherRow(
+    onRefresh: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .heightIn(min = 64.dp)
+            .padding(16.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Box(
+            modifier = Modifier
+                .size(48.dp)
+                .clip(CircleShape)
+                .background(amber600)
```

Comment:
Could this use a theme color?

Reply:
This is for the sun, so I didn't know how to handle it in the theme.